.github/workflows/hermetic_library_generation.yaml

```diff
@@ -17,24 +17,29 @@ name: Hermetic library generation upon generation config change through pull req
 on:
   pull_request:
 
+env:
+  REPO_FULL_NAME: ${{ github.event.pull_request.head.repo.full_name }}
+  GITHUB_REPOSITORY: ${{ github.repository }}
 jobs:
   library_generation:
-    # skip pull requests coming from a forked repository
-    if: github.event.pull_request.head.repo.full_name == github.repository
     runs-on: ubuntu-latest
     steps:
+    - name: Determine whether the pull request comes from a fork
+      run: |
+        if [[ "${GITHUB_REPOSITORY}" != "${REPO_FULL_NAME}" ]]; then
+          echo "This PR comes from a fork. Skip library generation."
+          echo "SHOULD_RUN=false" >> $GITHUB_ENV
+        else
+          echo "SHOULD_RUN=true" >> $GITHUB_ENV
+        fi
     - uses: actions/checkout@v4
+      if: env.SHOULD_RUN == 'true'
       with:
         fetch-depth: 0
         token: ${{ secrets.CLOUD_JAVA_BOT_TOKEN }}
-    - name: Generate changed libraries
-      shell: bash
-      run: |
-        set -x
-        [ -z "$(git config user.email)" ] && git config --global user.email "cloud-java-bot@google.com"
-        [ -z "$(git config user.name)" ] && git config --global user.name "cloud-java-bot"
-        bash .github/scripts/hermetic_library_generation.sh \
-          --target_branch ${{ github.base_ref }} \
-          --current_branch ${{ github.head_ref }}
-      env:
-        GH_TOKEN: ${{ secrets.CLOUD_JAVA_BOT_TOKEN }}
+    - uses: googleapis/sdk-platform-java/.github/scripts@v2.46.0
```

Comment:
Would there be another PR to group this with sdk-platform-java upgrade? Or maybe we can do it in this PR?

Reply:
Never mind, I remember it would be part of the renovate bot template change.